src/Scratch.jl

```diff
@@ -117,7 +131,13 @@ function track_scratch_access(pkg_uuid::UUID, scratch_path::AbstractString)
             if p.uuid == pkg_uuid
                 source_path = Base.pathof(m)
                 if source_path !== nothing
-                    return Base.current_project(dirname(source_path))
+                    project_path = ignore_eacces() do
+                        Base.current_project(dirname(source_path))
```

Comment:
The tracking code tries to attribute accesses to a particular project, so that when that project gets removed from disk, the scratchspace can also be removed.  In this case, if the owning module (TimeZones) has no on-disk project (because it's in a system image) this will fail.  We'll fall through to just using the `active_project` below, which is the only thing that makes sense.

Because we're using `Base.pathof()` to start `source_path`, I can't think of a realistic situation where `Base.pathof()` gives a real path, but there's a broken chain of directory permissions between the `.jl` file that was loaded and the project that owns that file.